grype/version/kb_version.go

```diff
@@ -0,0 +1,40 @@
+package version
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type kbVersion struct {
+	version string
+}
+
+func newKBVersion(raw string) kbVersion {
+	// XXX Is this even useful/necessary?
+	return kbVersion{
+		version: raw,
+	}
+}
+
+func (v *kbVersion) Compare(other *Version) (int, error) {
+	if other.Format != KBFormat {
+		return -1, fmt.Errorf("unable to compare kb to given format: %s", other.Format)
+	}
+	if other.rich.kbVer == nil {
+		return -1, fmt.Errorf("given empty kbVersion object")
+	}
+
+	return other.rich.kbVer.compare(*v), nil
+}
+
+// Compare returns 0 if v == v2, 1 otherwise
+func (v kbVersion) compare(v2 kbVersion) int {
+	if reflect.DeepEqual(v, v2) {
```

Comment:
clarifying question: why compare lexically when the known version formats are integers? I'm wondering about cases similar to `"99" > "100"` (where lexically this resolves to `True`, where the right answer is `False`).

Reply:
They look like integers but they might as well be a checksum ID because they don't represent an order. It would never be correct to attempt a comparison with `>`, or `<` operators

Reply:
gotcha, thanks 👍